.github/workflows/build.yml

```diff
@@ -6,20 +6,12 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
```

Comment:
There's a lot of duplication between this one and the macos build.

As far as I can tell, the duplication seems to be in order to avoid uploading artifacts on macos that we won't use? If so, it'd be nice to make the upload artifact step conditional so that it's only ran on linux. Code duplication is very error-prone, particularly inside yaml files, so it's good to be able to avoid it when possible.

Reply:
Splitting the `build` job like this was done to ensure that the `s2n-tests` job runs even if the `build-macos` job fails. I agree that this is less than ideal; if there's a better way to accomplish this, let's change it.

Reply:
The continue-on-error should've made it so that the s2n-tests job would run even if the macos jobs failed. Was that not the case?

Reply:
My impression was that it was allowing the job to complete (i.e. not immediately killing the Ubuntu jobs) but still skipping the dependent job. You can see this here: https://github.com/GaloisInc/saw-script/runs/1022813322. It's very possible that I've misinterpreted this, though.